ulib/experimental/Steel.Effect.Atomic.fst

```diff
@@ -85,7 +79,7 @@ let repr a framed opened_invariants f pre post req ens =
 
 let return a x opened_invariants #p = fun _ -> x
 
-#push-options "--fuel 0 --ifuel 0"
+#push-options "--fuel 0 --ifuel 1"
```

Comment:
This ifuel of 1 is needed for the new observability enum correct? As far as I remember, EverCrypt/HACL uses calls to `allow_inversion` on specific datatypes (typically, enums) to avoid this, could we adopt something similar here?

Reply:
See for instance https://github.com/project-everest/hacl-star/blob/927a70b03df989356ab449dad579d949fce8d116/providers/evercrypt/EverCrypt.Error.fsti#L14 , we could expose the same pattern in Steel.Effect.Atomic.fsti

Reply:
Not sure how important it is though since all of our existing code was verifying with this datatype abstract.

Reply:
Thanks for pointing it out! On the contrary, what do you think about removing this fuel setting altogether? The proofs go through even without it, so why use the voodoo magic when it is not necessary? HACL*/EverCrypt proofs are tricky with nl arithmetic etc., so it makes sense to control these options there, but our proofs are quite straightforward in comparison. Also I doubt if the clients are ever going to need to invert this type.